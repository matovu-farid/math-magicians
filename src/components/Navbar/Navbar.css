.nav-list {
  display: flex;
  justify-content: flex-end;

}

.nav-list li{
  list-style-type: none;
  padding: 10px;

}