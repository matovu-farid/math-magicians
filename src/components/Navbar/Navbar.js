import './Navbar.css';
import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <div className="nav-bar">
    <h1>Math Magicians</h1>
    <ul className="nav-list">
      <li>
        <NavLink to="/" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>
          Home
        </NavLink>
      </li>
      <li>
        <NavLink to="/calculator" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>
          Calculator
        </NavLink>
      </li>
      <li>
        <NavLink to="/quote" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>
          Quote
        </NavLink>
      </li>
    </ul>
  </div>
);

export default Navbar;
