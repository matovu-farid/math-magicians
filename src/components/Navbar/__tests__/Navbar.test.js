import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('navbar tests', () => {
  const MockNavBar = () => (
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
  it('renders correctly', () => {
    const tree = renderer.create(<MockNavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should display correct value', () => {
    render(<MockNavBar />);
    const links = screen.queryAllByRole('link');

    expect(links.length).toBe(3);
  });

  it('home link should have an active class on innitial page load', () => {
    render(<MockNavBar />);
    const homeLink = screen.queryByRole('link', { name: 'Home' });
    expect(homeLink).toHaveClass('active');
  });
  it('calculator link should not have an active class on innital page load', () => {
    render(<MockNavBar />);
    const homeLink = screen.queryByRole('link', { name: /calculator/i });
    expect(homeLink).not.toHaveClass('active');
  });
  it('calculator link should have an active class when clicked', () => {
    render(<MockNavBar />);
    const homeLink = screen.queryByRole('link', { name: /calculator/i });
    userEvent.click(homeLink);
    expect(homeLink).toHaveClass('active');
  });
});
