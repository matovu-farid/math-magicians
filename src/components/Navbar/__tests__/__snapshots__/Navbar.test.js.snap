// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar tests renders correctly 1`] = `
<div
  className="nav-bar"
>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="nav-list"
  >
    <li>
      <a
        aria-current="page"
        className="active navlink"
        href="/"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          version="1.1"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0.5l-8 8 1.5 1.5 1.5-1.5v6.5h4v-3h2v3h4v-6.5l1.5 1.5 1.5-1.5-8-8zM8 7c-0.552 0-1-0.448-1-1s0.448-1 1-1c0.552 0 1 0.448 1 1s-0.448 1-1 1z"
          />
        </svg>
        <span>
          Home
        </span>
      </a>
    </li>
    <li>
      <a
        className="inactive navlink"
        href="/calculator"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm2 .5v2a.5.5 0 0 0 .5.5h7a.5.5 0 0 0 .5-.5v-2a.5.5 0 0 0-.5-.5h-7a.5.5 0 0 0-.5.5zm0 4v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5zM4.5 9a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1zM4 12.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5zM7.5 6a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1zM7 9.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5zm.5 2.5a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1zM10 6.5v1a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1a.5.5 0 0 0-.5.5zm.5 2.5a.5.5 0 0 0-.5.5v4a.5.5 0 0 0 .5.5h1a.5.5 0 0 0 .5-.5v-4a.5.5 0 0 0-.5-.5h-1z"
          />
        </svg>
        <span>
          Calculator
        </span>
      </a>
    </li>
    <li>
      <a
        className="inactive navlink"
        href="/quote"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 2a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H4.414a1 1 0 0 0-.707.293L.854 15.146A.5.5 0 0 1 0 14.793V2zm7.194 2.766a1.688 1.688 0 0 0-.227-.272 1.467 1.467 0 0 0-.469-.324l-.008-.004A1.785 1.785 0 0 0 5.734 4C4.776 4 4 4.746 4 5.667c0 .92.776 1.666 1.734 1.666.343 0 .662-.095.931-.26-.137.389-.39.804-.81 1.22a.405.405 0 0 0 .011.59c.173.16.447.155.614-.01 1.334-1.329 1.37-2.758.941-3.706a2.461 2.461 0 0 0-.227-.4zM11 7.073c-.136.389-.39.804-.81 1.22a.405.405 0 0 0 .012.59c.172.16.446.155.613-.01 1.334-1.329 1.37-2.758.942-3.706a2.466 2.466 0 0 0-.228-.4 1.686 1.686 0 0 0-.227-.273 1.466 1.466 0 0 0-.469-.324l-.008-.004A1.785 1.785 0 0 0 10.07 4c-.957 0-1.734.746-1.734 1.667 0 .92.777 1.666 1.734 1.666.343 0 .662-.095.931-.26z"
          />
        </svg>
        <span>
          Quote
        </span>
      </a>
    </li>
  </ul>
</div>
`;
